app/Exports/LibraryExport.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Exports;
+
+use APP\Repositories\BookRepository;
+// use DebugBar\DebugBar;
+use Barryvdh\Debugbar\Facades\Debugbar;
+use Illuminate\Support\Collection;
+use Maatwebsite\Excel\Concerns\FromCollection;
+use Maatwebsite\Excel\Concerns\WithHeadings;
+use Maatwebsite\Excel\Concerns\WithTitle;
+use Maatwebsite\Excel\Concerns\ShouldAutoSize;
+use Illuminate\Support\Facades\Log;
+
+class LibraryExport implements FromCollection, WithTitle, WithHeadings, ShouldAutoSize
+{
+    
+    private $bookRepository;
+
+    public function __construct(BookRepository $bookRepository)
+    {
+        $this->bookRepository = $bookRepository;
+    }
+    
+    /**
+    * @return \Illuminate\Support\Collection
+    */
+    public function collection()
+    {
+        $books = $this->bookRepository->all(['*']);
+        $exportData = [];
+        foreach($books as $book) {
+            //possible here to do some data filtering and manipulation etc, but isn't necesary at time of writing
+            $book->makeHidden('updated_at', 'deleted_at', 'created_at');
+            $data = $book->toArray();
+            
+            array_push($exportData, $data);
+        }
+        return new Collection($exportData);
+    }
+    
+    public function title(): String
+    {
+        return 'Books';
+    }
+    
+    public function headings(): array
+    {
+        return [
+            '#',
+            'Title',
+            'Year',
+            'Type',
+            'Country',
+            'Code',
+        ];
+    }
```

Comment:
Since the header and actual sheet contents are defined separately (as is always the case with the laravel-excel package), it could happen that the sheet contents are not aligned with the header labels anymore at some point.

I'd suggest defining a separate constant that defines the properties to be exported, and which label to use in the header for each such property. Added benefit is that we can use that constant to limit which columns are included in the export.

[This](https://gist.github.com/marijnvanderhorst/c1a12bc33d671bb24aef906b942556bc) is an example of what the end result could look like (not tested).